### PR TITLE
Use xhyve to get the mac address

### PR DESCRIPTION
**Pull request by johanneswuerbach**
https://github.com/mist64/xhyve/pull/79 allows to fetch the mac address
using xhyve instead of using vmnet directly

### CODE REVIEW

File: xhyve.go [hunk 1]
```diff
@@ -596,3 +575,52 @@ func trimMacAddress(rawUUID string) string {
...
+		"-s", fmt.Sprintf("4,virtio-blk,%s", img),
+		"-f", fmt.Sprintf("kexec,%s,%s,%s", vmlinuz, initrd, bootcmd)}
+}
+
+func (d *Driver) getMACAdress() (string, error) {
+	args := append(d.xhyveArgs(), "-M")
+
+	stdout := bytes.Buffer{}
+
+	cmd := exec.Command(os.Args[0], args...) // TODO: Should be possible without exec
```
Comment (johanneswuerbach) [inline xhyve.go]: I couldn't find a way to get the stdout of xhyve without spawning. This works as a workaround for now.

**Comment by zchee** [comment]
@johanneswuerbach Thanks!!

but, If remove `vmnet/*`, We will not be able to test in travis for xhyve can not start.
See https://github.com/zchee/docker-machine-driver-xhyve/pull/37 and https://travis-ci.org/zchee/docker-machine-driver-xhyve/jobs/99313858

How do you think?

**Comment by zchee** [comment]
@johanneswuerbach 
Or, we do not need to be test `uuid` and `mac` ?

**Comment by johanneswuerbach** [comment]
While more coverage is great, I personally would prefer to merge this PR in this case and not test for `uuid` and `mac` for now. Copying at lot of xhyve src to be able to test feels worse and might also break in the future. After this PR this driver would still work when xhyve decides to change their UUID2Mac implementation.

WDYT?

**Comment by zchee** [comment]
@johanneswuerbach OK, I understand.
LGTM.

**Comment by johanneswuerbach** [comment]
I like the way virtualbox is unit tested in docker/machine, maybe something worth adapting for xhyve https://github.com/docker/machine/blob/master/drivers/virtualbox/virtualbox_test.go

**Comment by zchee** [comment]
@johanneswuerbach Thanks.
I will use that as reference from now on.

**Pull request merged**